src/loaders/compute_tools/tool_common.py

```diff
@@ -399,7 +403,14 @@ def _prepare_tool(
             suffix_ids,
         )
         if data_file:
-            meta[data_id] = {_META_TOOL_IDENTIFIER: tool_identifier, _META_SOURCE_FILE: data_file}
+
+            mata_filename_path = os.path.join(data_file.parent, data_file.parent.name + ".meta")
+            meta_filename = os.path.basename(mata_filename_path) if os.path.exists(mata_filename_path) else ""
+
+            meta[data_id] = {_META_TOOL_IDENTIFIER: tool_identifier, 
+                             _META_SOURCE_FILE: data_file,
+                             _META_SOURCE_DIR: data_file.parent,
+                             _META_SOURCE_FILE_NAME: meta_filename}
```

Comment:
so we are using full path for the source assembly file, but only the file name for the meta file. We should keep them consistent.